Pod/API/Internal/Parse.swift

```diff
@@ -0,0 +1,139 @@
+     
+     - parameter method:              The method to call
+     - parameter parameters:          The parameters of the call, if any
+     - parameter responseArrayOfType: The type of item expected inside the response array
+     - parameter responsePostProcess: A function to apply to the parsed response array before invoking the callback
+     - parameter callback:            The callback to execute with the result of the call
+     */
+    func execute<ResponseItemType: XMLRPCDeserializable>(
+        method: Method,
+        parameters: [String: Any]? = nil,
+        responseArrayOfType: ResponseItemType.Type,
+        responsePostProcess: ([ResponseItemType]) -> ([ResponseItemType]?) = { $0 },
+        callback: Of<[ResponseItemType]>.callback) {
+            self.execute(
+                method,
+                parameters: parameters,
+                responseType: ArrayContainer<ResponseItemType>.self,
+                responsePostProcess: { arrayContainer in
+                    guard let processedArray = responsePostProcess(arrayContainer.array) else {
+                        return nil
+                    }
+                    return ArrayContainer<ResponseItemType>(array: processedArray)
+                },
+                callback: { result in
+                    if let value = result.value where result.isSuccess {
+                        callback(.Success(value.array))
+                    } else {
+                        callback(.Failure(result.error!))
+                    }
+            })
+    }
+    
+}
+
+extension XMLRPCNode {
+    /// If the node has a string value, return that; otherwise, if the node has a base64 value, assume it's UTF8 data and convert that to a string.
+    var ljString: String? {
+        if let string = self.string {
+            return string
+        }
+        guard let data = self.data else {
+            return nil
+        }
+        return String(data: data, encoding: NSUTF8StringEncoding)
+    }
+    
+    /**
+     Convert a node to an array with items of a known type
+     
+     - parameter type: The type of the items in the array
+     
+     - throws: `API.ParseError.BadStructureInArray` if any item in the array failed to parse; `UnwrapError.Nil` if the node is not an array
+     
+     - returns: An array of items of the given type
+     */
+    func arrayOfType<ItemType: XMLRPCDeserializable>(type: ItemType.Type) throws -> [ItemType] {
+        let arrayOfNodes = try ¿self.array
+        let arrayOfItems = arrayOfNodes.flatMap({ try? ItemType(xmlrpcNode: $0) })
+        guard arrayOfNodes.count == arrayOfItems.count else {
+            throw API.ParseError.BadStructureInArray
+        }
+        return arrayOfItems
+    }
+}
+
+/**
+ *  Helper to parse array responses.
+ */
+private struct ArrayContainer<T: XMLRPCDeserializable> {
+    let array: [T]
+}
+extension ArrayContainer: XMLRPCDeserializable {
+    init(xmlrpcNode value: XMLRPCNode) throws {
```

Comment:
... and here's a hacky workaround that I'm going to commit because it allows me to completely nuke the array-based `.execute` method:
```swift
extension Array: XMLRPCDeserializable {
    init(xmlrpcNode value: XMLRPCNode) throws {
        assertionFailure("Array with a non-XMLRPCDeserializable element type isn't really XMLRPCDeserializable.")
        throw API.Error.Parsing
    }
}
extension Array where Element: XMLRPCDeserializable {
    init(xmlrpcNode value: XMLRPCNode) throws {
        let key = try ¿value.dictionary?.keys.first
        self.init(try ¿value[key].arrayOfType(Element))
    }
}
```